docs/DafnyRef/Specifications.md

```diff
@@ -601,61 +635,80 @@ interprets JML's `assigns/modifies` in sense (b).
 ACSL and ACSL++ use the `assigns` keyword, but with _modify_ (b) semantics.
 Ada/SPARK's dataflow contracts encode _write_ (a) semantics.
 
-### 5.1.7. Invariant Clause {#sec-invariant-clause}
-````grammar
-InvariantClause_ =
-  "invariant" { Attribute }
-  Expression(allowLemma: false, allowLambda: true)
-````
+### 19.1.7. Invariant Clause ([grammar](#g-invariant-clause)) {#sec-invariant-clause}
+
+Examples:
+<!-- %check-resolve-warn Specifications.2.expect -->
+```dafny
+method m()
+{
+  var i := 10;
+  while 0 < i
+    invariant 0 <= i < 10
+}
+```
 
 An `invariant` clause is used to specify an invariant
 for a loop. If more than one `invariant` clause is given for
 a loop the effective invariant is the conjunction of
-the conditions specified.
+the conditions specified, in their textual order.
```

Comment:
What is a textual order?

Reply:
The order in the source text, in case previous ones are needed to establish later ones or to ensure the later ones are well-defined. Same as for preconditions and for postconditions

Reply:
Yes, thank you.